core/src/main/java/net/adoptopenjdk/icedteaweb/client/controlpanel/ControlPanel.java

```diff
@@ -163,12 +163,13 @@ private JPanel createButtonPanel() {
      * 
      * @return A panel with all the components in place.
      */
-    private JPanel createMainSettingsPanel() {
+    private JPanel createMainSettingsPanel(final ControlPanelStyle style) {
+        Assert.requireNonNull(style, "style");
         final Map<String, ControlPanelProvider> providers = new HashMap<>();
         final ServiceLoader<ControlPanelProvider> serviceLoader = ServiceLoader.load(ControlPanelProvider.class);
         serviceLoader.iterator().forEachRemaining(p -> {
             final String name = p.getName();
-            if (p.isActive()) {
+            if (p.isActive() && style.isPanelActive(p.getName())) {
```

Comment:
is the p.isActive() still needed?
The extension can also not register the service to the service loader which would have the same effect...